VideoLocker/src/main/java/org/edx/mobile/model/course/BlockModel.java

```diff
@@ -51,6 +51,6 @@
     public BlockData data;
 
     public boolean isContainer(){
-        return descendants != null && descendants.size() > 0;
+        return descendants != null;
```

Comment:
I don't really like using null as a sentinel if we can avoid it (since it's nice to be able to use empty objects to avoid crashes around null references). And I don't think the server makes any guarantees about whether children is nil or just the empty list.